devices/src/virtio/block.rs

```diff
@@ -725,7 +757,8 @@ mod tests {
         assert!(vq.end().0 < 0x1000);
 
         let queues = vec![vq.create_queue()];
-        let disk_image = b.disk_image.take().unwrap();
+        let mut disk_image = b.disk_image.take().unwrap();
+        let disk_nsectors = disk_image.seek(SeekFrom::End(0)).unwrap() / SECTOR_SIZE;
```

Comment:
I like the changes, good catch!

Please add a test that fails on an out-of-bounds read/write. Ideally also add a test with invalid `len` on the buffer for `GetDeviceID`.

Reply:
Done. Added a test for GetDeviceID failure. Fixed the sector out of bounds test.